libs/base/Data/List1.idr

```diff
@@ -164,6 +165,16 @@ export
 consInjective : (x ::: xs) === (y ::: ys) -> (x === y, xs === ys)
 consInjective Refl = (Refl, Refl)
 
+%hint
+export
+consHeadInj : {x : a} -> Injective (x :::)
+consHeadInj = MkInjective (\Refl => Refl)
```

Comment:
Is there a reason for naming these?

Reply:
Not particularly, I can try to implement them as `implementation` if that's what you're asking